naga/src/back/msl/writer.rs

```diff
@@ -3801,11 +3803,12 @@ impl<W: Write> Writer<W> {
             );
 
             // Is any global variable used by this entry point dynamically sized?
-            let supports_array_length = module
-                .global_variables
-                .iter()
-                .filter(|&(handle, _)| !fun_info[handle].is_empty())
-                .any(|(_, var)| needs_array_length(var.ty, &module.types));
+            let supports_array_length = pipeline_options.experimental_vertex_pulling_transform
```

Comment:
We should rename this var.

Reply:
Renamed to `needs_buffer_sizes`, here and in other places.